ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSwindow.cpp

```diff
@@ -694,4 +703,79 @@ bool clipboard_has_text()
   return value;
 }
 
+void clipboard_set_sprite(int ind, unsigned subimg) {
+  unsigned char *src = nullptr;
+  unsigned width, height;
+
+  enigma::sprite *spr;
+  if (!enigma::get_sprite_mtx(spr, ind)) return;  
+  src = enigma::graphics_copy_texture_pixels(spr->texturearray[subimg], &width, &height);
+
+  HBITMAP hBitmap = CreateBitmap(width, height, 1, 32, src);
+
+  OpenClipboard(NULL);
+  EmptyClipboard();
+  SetClipboardData(CF_BITMAP, hBitmap);
+  CloseClipboard();
+
+  CloseHandle(hBitmap);
+  free(src);
+}
+
+int clipboard_get_sprite(bool precise, bool transparent, bool smooth, int x_offset, int y_offset) {
+  if (!clipboard_has_sprite()) return -1;
+  OpenClipboard(NULL);
+  HBITMAP hBitmap = (HBITMAP)GetClipboardData(CF_BITMAP);
+  CloseClipboard();
+
+  BITMAPINFOHEADER bih = {0};
```

Comment:
Lock the bitmap data, and pass it to `image_decode_bmp`. You will need to construct a string or string view from the data and its size (eg, `string(clip_data, GlobalSize(clip_data))`).